documentation/SolutionsRegistryFormat.md

```diff
@@ -0,0 +1,194 @@
+
+**Example blocks**:
+```
+"configure": [
+    "settings.myconf"
+],
+"restore": [
+    "settings.myconf"
+],
+"start": [
+    {
+        "type": "gpii.launch.exec",
+        "command": "\"${{registry}.HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\nvda.exe\\}\""
+    }
+],
+"stop": [
+    {
+        "type": "gpii.windows.killProcessByName",
+        "filename": "nvda.exe"
+    }
+]
+```
+
+### update
+
+The `update` block works very similarly to the configure, restore, start and stop blocks. It describes what should happen when the configuration needs to be updated (eg. due to context changes, PCP adjustments, etc).
+
+The format of the `update` block allows for the same entries as the configure, restore, start and stop blocks - that is: arbitrary lifecycle action blocks and `settings.<blockname>`. Unlike for the other lifecycle blocks, the `update` block furthermore allows references to the `start`, `stop` and `configure` blocks. This is one by putting a string with the name of that block. When the system encounters one of these references, the entries of that block will be run.
+
+**Example block**:
+```
+"configure": [
+    "settings.myconf"
+],
+"update": [
+    "configure",
+    {
+        "type": "gpii.launch.exec",
+        "command": "my_application --refresh"
+    }
+]
+```
+
+In the above example, the process of updating the application settings would consists of running the contents of the `configure` block (that is `"settings.myconf"`), followed by a custom lifecycle actions.
+
+
+### isInstalled:
+
+This directive is used to detect whether a solution is installed. If any of these blocks evaluate to `true` (implicit **OR**), the application is considered to be installed.
+
+**Example Entry**:
+```
+"isInstalled": [
+    {
+        "type": "gpii.reporter.fileExists",
+        "fileName": "${{registry}.HKEY_CURRENT_USER\\Software\\Texthelp\\Read&Write10\\InstallPath}\\ReadAndWrite.exe"
+    }, // SEEMS LIKE IMPLICIT OR BETWEEN THESE BLOCKS
+    {
+        "type": "gpii.packageKit.find",
+        "name": "orca"
+    }
+]
+```
+
+### NOT IMPLEMENTED blocks:
+There are several advanced options that we're not planning to implement in the short term, but which will make the implementation of things like the ORCA settings handler much less horrible.
+
+#### isRunning:
+To detect whether a solution is running - this is planned to be integrated in the relatively short run. If any of these blocks evaluate to `true` (implicit OR), the application is considered to be running.
+
+**Example Entry**:
+```
+"isRunning": [
+    {
+        "type": "gpii.processReporter.byName",
+        "fileName": "firefox.exe"
+    },
+    {
+       "type": "gpii.processReporter.findProcessesByCommand",
+       "command": "ps aux | grep 'myprocess | wc -l",
```

Comment:
@kaspermarkus 
Nit:  missing close quote after myprocess.  I think you mean:
ps aux | grep 'myprocess' | wc -l